### PR TITLE
Add MaxUnpooling2DV2 layer

### PR DESCRIPTION
**Pull request by midsterx**
# Description

Add MaxUnpooling2DV2 layer.
The _max_pool_with_argmax_ function can map several input sizes to the same output size. Hence, to avoid ambiguity in the inversion process, the stride and padding arguments have been replaced by the output_size argument. This has been done in a new layer rather than in the existing MaxUnpooling2D class to avoid compatibility issues.

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [x] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [x] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

I run the layer_test

### CODE REVIEW

**Comment by midsterx** [comment]
@tensorflow/sig-addons-maintainers - kindly request one of you to take a look at this PR.

File: tensorflow_addons/layers/max_unpooling_2d_v2.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
```
Comment (thaink) [inline tensorflow_addons/layers/max_unpooling_2d_v2.py:1]: 2021

File: tensorflow_addons/layers/tests/max_unpooling_2d_v2_test.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
```
Comment (thaink) [inline tensorflow_addons/layers/tests/max_unpooling_2d_v2_test.py:1]: 2021

**Review by thaink — APPROVED** [review]
Thanks for the PR.

**Comment by thaink** [comment]
@WindQAQ @seanpmorgan Can you take a look at this PR?

File: tensorflow_addons/layers/max_unpooling_2d_v2.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+from typing import Iterable
+
+from tensorflow_addons.utils.keras_utils import normalize_tuple
+
+
+def _max_unpooling_2d_v2(updates, mask, output_size):
+    """Unpool the outputs of a maximum pooling operation."""
+    output_size_attr = " ".join(["i: %d" % v for v in output_size])
+    experimental_implements = [
+        'name: "addons:MaxUnpooling2DV2"',
```
Comment (fsx950223) [inline tensorflow_addons/layers/max_unpooling_2d_v2.py:29]: Is it available?

**Comment by fsx950223** [comment]
Thanks for the PR.

File: tensorflow_addons/layers/max_unpooling_2d_v2.py (hunk 3 above)
Comment (midsterx) [reply tensorflow_addons/layers/max_unpooling_2d_v2.py]: Thank you for catching this @fsx950223.
It is not available, and I have removed it from the PR.

File: tensorflow_addons/layers/tests/max_unpooling_2d_v2_test.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+# ==============================================================================
+"""Tests for MaxUnpooling2DV2 layers."""
+
+import numpy as np
+import pytest
+import tensorflow as tf
+from tensorflow_addons.layers.max_unpooling_2d_v2 import MaxUnpooling2DV2
+
+
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
```
[the remark this answers is not in the document]
Comment (midsterx) [reply tensorflow_addons/layers/tests/max_unpooling_2d_v2_test.py:23]: Could you provide some clarity regarding this?
Did you mean something like `@pytest.mark.with_device(["cpu", "gpu", tf.distribute.MirroredStrategy])`?

File: tensorflow_addons/layers/tests/max_unpooling_2d_v2_test.py (hunk 2 above)
Comment (midsterx) [reply tensorflow_addons/layers/tests/max_unpooling_2d_v2_test.py]: Thank you for pointing this @thaink, I have corrected the year.

**Review by fsx950223 — DISMISSED** [review]

**Comment by fsx950223** [comment]
You should update codeowner too.

**Comment by midsterx** [comment]
> You should update codeowner too.

I added myself to .github/CODEOWNERS.
Thanks!

**Review by fsx950223 — APPROVED** [review]

**Pull request merged**